pkg/ec2provider/ec2provider.go

```diff
@@ -0,0 +1,274 @@
+package ec2provider
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/httputil"
+	"sync"
+	"time"
+)
+
+const (
+	// max limit of k8s nodes support
+	max_channel_size = 5000
+	// max number of in flight non batched ec2:DescribeInstances request to flow
+	max_allowed_single_request = 5
+	// default wait interval for the ec2 instance id request which is already in flight
+	default_wait_interval = 50
+	// Making sure the single instance calls waits max till 5 seconds 100* (50 * time.Millisecond)
+	total_iteration_for_wait_interval = 100
```

Comment:
Isn't 5 seconds quite a long time?

Reply:
5 seconds is the timeout for any given request and upper limit.